davmail: 5.4.0 -> 5.5.1

dguibert opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

dguibert committed:
davmail: 5.4.0 -> 5.5.1

r-rmcgibbo commented:
Result of `nixpkgs-review pr 122856` at aae11f33 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>davmail</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/davmail/default.nix:15:3:
```
   |
15 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/davmail/default.nix:28:5:
```
   |
28 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

dguibert committed:
davmail: add runHook preInstall and runHook postInstall

Apply hint from nixpkgs-hammer: missing-phase-hooks

installPhase should probably contain runHook preInstall and runHook
postInstall.

dguibert committed:
davmail: gpl2 -> gpl2Plus

Apply hint form nixpkgs-hammer: unclear-gpl

gpl2 is a deprecated license, please check if project uses gpl2Plus or
gpl2Only and change meta.license accordingly.
